sdks/python/apache_beam/io/gcp/spannerio.py

```diff
@@ -0,0 +1,333 @@
+      project_id: The ID of the project which owns the instances, tables
+        and data.
+      instance_id: The ID of the instance.
+      database_id: The ID of the database instance.
+      user_agent: (Optional) The user agent to be used with API request.
+      pool: (Optional) session pool to be used by database.
+      read_timestamp: (Optional) Execute all reads at the given timestamp.
+      exact_staleness: (Optional) Execute all reads at a timestamp that is
+        ``exact_staleness`` old.
+    """
+    warnings.warn("ReadFromSpanner is experimental.", FutureWarning,
+                  stacklevel=2)
+    self._transaction = None
+    self._options = _BeamSpannerConfiguration(
+        project=project_id, instance=instance_id, database=database_id,
+        credentials=credentials, user_agent=user_agent, pool=pool,
+        snapshot_read_timestamp=read_timestamp,
+        snapshot_exact_staleness=exact_staleness
+    )
+
+  def with_query(self, sql, params=None, param_types=None):
+    read_operation = [ReadOperation.with_query(sql, params, param_types)]
+    return self.read_all(read_operation)
+
+  def with_table(self, table, columns, index="", keyset=None):
+    read_operation = [ReadOperation.with_table(
+        table=table, columns=columns, index=index, keyset=keyset
+    )]
+    return self.read_all(read_operation)
+
+  def read_all(self, read_operations):
+    if self._transaction is None:
+      return _BatchRead(read_operations=read_operations,
+                        spanner_configuration=self._options)
+    else:
+      return _NaiveSpannerRead(transaction=self._transaction,
+                               read_operations=read_operations,
+                               spanner_configuration=self._options)
+
+  @staticmethod
+  @experimental(extra_message="(ReadFromSpanner)")
+  def create_transaction(project_id, instance_id, database_id, credentials=None,
+                         user_agent=None, pool=None, read_timestamp=None,
+                         exact_staleness=None):
+    """
+    Return the snapshot state for reuse in transaction.
+
+    Args:
+      project_id: The ID of the project which owns the instances, tables
+        and data.
+      instance_id: The ID of the instance.
+      database_id: The ID of the database instance.
+      credentials: (Optional) The OAuth2 Credentials to use for this client.
+      user_agent: (Optional) The user agent to be used with API request.
+      pool: (Optional) session pool to be used by database.
+      read_timestamp: (Optional) Execute all reads at the given timestamp.
+      exact_staleness: (Optional) Execute all reads at a timestamp that is
+        ``exact_staleness`` old.
+      """
+    _snapshot_options = {}
+    if read_timestamp:
+      _snapshot_options['read_timestamp'] = read_timestamp
+    if exact_staleness:
+      _snapshot_options['exact_staleness'] = exact_staleness
+
+    spanner_client = Client(project=project_id, credentials=credentials,
```

Comment:
Usually, the user would call this method only one and if he is calling this method again for some reason, he would be assuming that it will create a new session and transaction! 
What're your thoughts?